packages/daemon/src/host.js

```diff
@@ -514,12 +523,39 @@ export const makeHostMaker = ({
       return cancelValue(formulaIdentifier, reason);
     };
 
+    /** @type {import('./types.js').EndoHost['gateway']} */
+    const gateway = async () => {
```

Comment:
I like where this is headed. Maybe leave a note that this corresponds to the nonce locator from E. Similar names include entrance, threshold, door, welcome.